src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SubSourceLogic.cs

```diff
@@ -488,6 +488,22 @@ private void HandleSubsourceCancelled((TopicPartition, ISubSourceCancellationStr
         _subSources = _subSources.Remove(topicPartition);
         _partitionsInStartup = _partitionsInStartup.Remove(topicPartition);
 
+        // If the stage is shutting down, don't re-emit partitions to avoid infinite loops
+        if (_shutdownTokenSource.IsCancellationRequested)
+        {
+            if (Log.IsDebugEnabled)
+                Log.Debug("#{0} Stage is shutting down, not re-emitting partition {1}", _actorNumber, topicPartition);
+            return;
+        }
+
+        // Check if this partition is scheduled for revocation - if so, don't re-emit
+        if (_partitionsToRevoke.Contains(topicPartition))
```

Comment:
If we're not shutting down, but this partition is being revoked, don't re-emit it